server/src/main/java/com/broadcom/lsp/cobol/core/messages/JSONResourceBundleControl.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package com.broadcom.lsp.cobol.core.messages;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.List;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+/** This class is an extension of {@link ResourceBundle.Control} to support JSON objects. */
+public class JSONResourceBundleControl extends ResourceBundle.Control {
+
+  public static final String DEFAULT_LOCALE = "en";
+  private static final String SUPPORTED_RESOURCE_BUNDLE_EXTENSIONS = "json";
+
+  /**
+   * Instantiates a resource bundle for the given bundle name of the given format and locale, using
+   * the given class loader if necessary. This method returns <code>null</code> if there is no
+   * resource bundle available for the given parameters. If a resource bundle can't be instantiated
+   * due to an unexpected error, the error must be reported by throwing an <code>Error</code> or
+   * <code>Exception</code> rather than simply returning <code>null</code>.
+   *
+   * <p>If the <code>reload</code> flag is <code>true</code>, it indicates that this method is being
+   * called because the previously loaded resource bundle has expired.
+   *
+   * <p>For more info refer to java docs {@link ResourceBundle.Control#newBundle(String, Locale,
+   * String, ClassLoader, boolean)}
+   *
+   * @param baseName the base bundle name of the resource bundle, a fully qualified class name
+   * @param locale the locale for which the resource bundle should be instantiated
+   * @param format the resource bundle format to be loaded
+   * @param loader the <code>ClassLoader</code> to use to load the bundle
+   * @param reload the flag to indicate bundle reloading; <code>true</code> if reloading an expired
+   *     resource bundle, <code>false</code> otherwise
+   * @return the resource bundle instance, or <code>null</code> if none could be found.
+   * @throws NullPointerException if <code>bundleName</code>, <code>locale</code>, <code>format
+   *     </code>, or <code>loader</code> is <code>null</code>, or if <code>null</code> is returned
+   *     by {@link #toBundleName(String, Locale) toBundleName}
+   * @throws IOException if an error occurred when reading resources using any I/O operations
+   */
+  @Override
+  public ResourceBundle newBundle(
+      String baseName, Locale locale, String format, ClassLoader loader, boolean reload)
```

Comment:
Why you throw checked exception?  Should be handled here rather than than in used place.

Reply:
Also I was expecting, all the messages in the application are extracted and replaced with appropriate place. That is the intention behind the task.

Reply:
Optional: I still recommend to use simple Java ResourceBundle way with properties file, just will end-up with one util-class. Here it looks a lot implementations just to read and parsing to reach same functionality. Our aim is just to externalize raw string messages.

Reply:
1. Ideally, we will never enter to this condition. And even if we do, we can only manage locale. And locale is something we have/should be managed in the message service (the object creating the resourceBundle). I have tried to keep the concern to manage locale outside of this class as this class is only responsible to create a resource bundle. Actually, the real code for a locale is something that I am trying now (get from the client). But facing certain issues. I might reach out to you .

2.
`messageService.getMessage("some key","param1","param2").getFormattedMessage()`
 would actually give replaced string.   

3. Now I think the same. I may have gone way over the top in the excitement of 1st task .

Reply:
Sure, just ping me